addons/folo/common/src/main/java/org/commonjava/indy/folo/data/FoloRecordCassandra.java

```diff
@@ -0,0 +1,426 @@
+
+        logger.warn("-- Cassandra Folo Records Keyspace and Tables created  ---");
+
+    }
+
+    @Override
+    public boolean recordArtifact(TrackedContentEntry entry) throws FoloContentException, IndyWorkflowException {
+
+        String buildId = entry.getTrackingKey().getId();
+        String path = entry.getPath();
+
+//        String api = uriInfo.getBaseUriBuilder().path("api").build().toString();
+        String content  = "content";
+        String packageType = entry.getStoreKey().getPackageType();
+        String storeKey = entry.getStoreKey().getType().singularEndpointName();
+        String name = entry.getStoreKey().getName();
+
+
+        try {
+            String localUrl;
+
+            localUrl = UrlUtils.buildUrl("" ,content , packageType , storeKey ,name , path );
+
+            BoundStatement bind = getTrackingRecordByBuildIdAndPath.bind(buildId,path);
+            ResultSet trackingRecord = session.execute(bind);
+            Row one = trackingRecord.one();
+
+            if(one!=null) {
+                DtxTrackingRecord dtxTrackingRecord = one.get(0, DtxTrackingRecord.class);
+                Boolean state = dtxTrackingRecord.getState();
+
+                if(state) {
+                    throw new FoloContentException( "Tracking record: {} is already sealed!", entry.getTrackingKey() );
+                }  else {
+
+                    DtxTrackingRecord dtxTrackingRecord1 =
+                            DtxTrackingRecord.fromTrackedContentEntry(entry,false);
+
+                    if(dtxTrackingRecord.getTrackingKey().equals(dtxTrackingRecord1.getTrackingKey()) &&
+                        dtxTrackingRecord.getPath().equals(dtxTrackingRecord1.getPath())) {
+
+                        dtxTrackingRecord1.setLocalUrl(localUrl);
+
+                        trackingMapper.save(dtxTrackingRecord1);
+                        return true;
+                    } else {
+                        return false;
+                    }
+                }
+
+            } else {
+                DtxTrackingRecord dtxTrackingRecord = new DtxTrackingRecord(entry);
+                dtxTrackingRecord.setLocalUrl(localUrl);
+                trackingMapper.save(dtxTrackingRecord); //  optional Options with TTL, timestamp...
+                return true;
+            }
+
+        } catch (MalformedURLException e) {
+            throw new IndyWorkflowException( "Mailformed URL  for record: {}  path: {}", entry.getTrackingKey(),entry.getPath() );
+        }
+
+
+    }
+
+    @Override
+    public void delete(TrackingKey key) {
+
+        // Do we  want to remove records from DB?
+
+    }
+
+    @Override
+    public void replaceTrackingRecord(TrackedContent record) {
+
+        saveTrackedContentRecords(record);
+
+    }
+
+    @Override
+    public boolean hasRecord(TrackingKey key) {
+        return hasSealedRecord(key) || hasInProgressRecord(key);
+    }
+
+    @Override
+    public boolean hasSealedRecord(TrackingKey key) {
```

Comment:
This method isn't used anywhere except in a test, and the `hasRecord()` method. We should definitely make sure we're **NOT** optimizing for this method call, and we should take this out.

Reply:
I agree. will be removed in next commit.